optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -251,14 +251,23 @@
     <GroupKeyA_, GroupKeyB_> BiConstraintStream<GroupKeyA_, GroupKeyB_> groupBy(
             BiFunction<A, B, GroupKeyA_> groupKeyAMapping, BiFunction<A, B, GroupKeyB_> groupKeyBMapping);
 
-    /*
-    // TODO implement this
-    <GroupKeyA_, GroupKeyB_, ResultContainer_, Result_>
-    TriConstraintStream<GroupKeyA_, GroupKeyB_, Result_> groupBy(
-            BiFunction<A, B, GroupKeyA_> groupKeyAMapping,
-            BiFunction<A, B, GroupKeyB_> groupKeyBMapping,
-            BiConstraintCollector<A, B, ResultContainer_, Result_> collector);
+    /**
+     * Combines the semantics of {@link #groupBy(BiFunction, BiFunction)} and {@link #groupBy(BiConstraintCollector)}.
+     * That is, the first and second facts in the tuple follow the {@link #groupBy(BiFunction, BiFunction)} semantics,
+     * and the third fact is the result of applying {@link BiConstraintCollector#finisher()} on all the tuples of the
+     * original {@link UniConstraintStream} that fall in the group.
+     * @param groupKeyAMapping never null, function to convert first fact in the original tuple to a different fact
+     * @param groupKeyBMapping never null, function to convert second fact in the original tuple to a different fact
```

Comment:
- missing "the" before "first fact"
- These 2 lines are incorrect:
"... to convert the first fact in the original tuple ..."
Each groupMaping converts the entire tuple. They both get A *and* B as input.
=> "groupKeyAMapping never null, function to convert the original tupel into a first fact"
"groupKeyBMapping never null, function to convert the original tupel into a second fact"